pkg/services/ngalert/api/compat.go

```diff
@@ -223,11 +224,31 @@ func AlertRuleExportFromAlertRule(rule models.AlertRule) (definitions.AlertRuleE
 		result.Annotations = &rule.Annotations
 	}
 	if rule.Labels != nil {
-		result.Labels = &rule.Labels
+		result.Labels = escapeLabelsForFileProvisioning(rule.Labels)
 	}
 	return result, nil
 }
 
+// this is to prevent the string from being interpreted as a variable in the provisioning engine
+func escapeLabelsForFileProvisioning(labels map[string]string) *map[string]string {
+	escapedLabels := make(map[string]string)
+	for k, v := range labels {
+		escapedLabels[k] = addEscapeCharactersToString(v)
+	}
+	return &escapedLabels
+}
+
+func addEscapeCharactersToString(s string) string {
+	// loop over words in the string and add escape characters to any word that starts with $
+	words := strings.Split(s, " ")
+	for i, word := range words {
+		if strings.HasPrefix(word, "$") {
+			words[i] = "$" + word
+		}
+	}
+	return strings.Join(words, " ")
+}
```

Comment:
this logic skips strings like `test$A` as well as `$$$ABC`. I think the logic should be more straightforward and just replace every `$` with `$$`.

Reply:
I think the environment templating on the file provisioning side doesn't replace those kinds of strings but I will double check the behavior. My goal was to get the same output that would when provided as input to file provisioning would result in a consistent result. Export -> import -> export should remain consistent. I think if I replace every `$` then we'd get an addition of `$` with every export.